### PR TITLE
AlwaysOffSampler

### PR DESCRIPTION
**Pull request by nholbrook**

### CODE REVIEW

File: sdk/test/trace/always_off_sampler_test.cc [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+{}
+
+TEST(AlwaysOffSampler, MockSpanContext)
+{
+	Sampler sampler(new AlwaysOffSampler());
+
+	ASSERT_EQ("AlwaysOffSampler", sampler.GetDescription());
+
+	// Placeholder
+	MockSpanContext context(new MockSpanContext());
```
Comment (oulin-coder) [inline sdk/test/trace/always_off_sampler_test.cc]: You don't need to pass new MockSpanContext() when instantiating it on the stack. You can just leave it as 

MockSpanContext context();

File: sdk/test/trace/always_off_sampler_test.cc [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+	constexpr uint8_t buf[] = {1, 2, 3, 4, 5, 6, 7, 8, 8, 7, 6, 5, 4, 3, 2, 1}
```
Comment (oulin-coder) [inline sdk/test/trace/always_off_sampler_test.cc]: Add a comment for why buf is initialized with these values?

File: sdk/test/trace/always_off_sampler_test.cc [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+using namespace opentelemetry::sdk::trace;
+
+// Placeholder
+class MockSpanContext
+{}
+
+TEST(AlwaysOffSampler, MockSpanContext)
+{
+	Sampler sampler(new AlwaysOffSampler());
```
Comment (oulin-coder) [inline sdk/test/trace/always_off_sampler_test.cc]: You can directly initialize AlwaysOffSampler.

File: sdk/src/trace/always_off_sampler.cc [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+  nostd::span<AttributeKeyValue> GetAttributes() const noexcept override
+  {
+    return attributes_;
+  }
+
+private:
+  Decision decision_ = Decision::NOT_RECORD;
+  nostd::span<AttributeKeyValue> attributes_ = nostd::span();
+};
```
Comment (oulin-coder) [inline sdk/src/trace/always_off_sampler.cc]: FYI there's been some updates to sampling result.

**Review by oulin-coder — COMMENTED** [review]
PTAL updated SamplingResult, otherwise looking good.

File: sdk/test/trace/always_off_sampler_test.cc [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+	// Placeholder
+	MockSpanContext context(new MockSpanContext());
+
+	constexpr uint8_t buf[] = {1, 2, 3, 4, 5, 6, 7, 8, 8, 7, 6, 5, 4, 3, 2, 1}
+	opentelemetry::trace::TraceId trace_id(buf);
+
+	auto sampling_result = sampler.ShouldSample(
+		context, trace_id, "Test", SpanKind.kInternal, nostd::span());
+
+	ASSERT_NE(nullptr, sampling_result);
```
Comment (ziqizh) [inline sdk/test/trace/always_off_sampler_test.cc]: can't compare a class with a null pointer

File: sdk/test/trace/always_off_sampler_test.cc [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+	ASSERT_EQ("AlwaysOffSampler", sampler.GetDescription());
+
+	// Placeholder
+	MockSpanContext context(new MockSpanContext());
+
+	constexpr uint8_t buf[] = {1, 2, 3, 4, 5, 6, 7, 8, 8, 7, 6, 5, 4, 3, 2, 1}
+	opentelemetry::trace::TraceId trace_id(buf);
+
+	auto sampling_result = sampler.ShouldSample(
+		context, trace_id, "Test", SpanKind.kInternal, nostd::span());
```
Comment (ziqizh) [inline sdk/test/trace/always_off_sampler_test.cc]: Use `SpanKind::kInternal` to access an enum.

**Review by ziqizh — COMMENTED** [review]
- Please consider merge branch `sampling-api-header' as there might be small changes in the header file.
- Consider adding a `cc_test` in the BUILD file.

**Pull request closed without merging**